File: Backend/carShop/src/main/java/com/contrader/carShop/dto/ConcessionariaDTO.java
package com.contrader.carShop.dto;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import com.contrader.carShop.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
public class ConcessionariaDTO {

    private Long id;

    private String nome;

    private String descrizione;

    private String provincia;

    private String citta;

    private String indirizzo;

    private String nazione;

    private User user;
}

File: Frontend/src/dto/concessionariadto.ts
import { UserDTO } from "./userdto";

export class ConcessionariaDTO {
    id: number;
    nome: String;
    descrizione: String;
    provincia: String;
    citta: String;
    indirizzo: String;
    nazione: String;
    user: UserDTO;

    constructor(id: number,
                nome: String,
                descrizione: String,
                provincia: String,
                citta: String,
                indirizzo: String,
                nazione: String,
                user: UserDTO){
        this.id = id;
        this.nome = nome;
        this.descrizione = descrizione;
        this.provincia = provincia;
        this.citta = citta;
        this.indirizzo = indirizzo;
        this.nazione = nazione;
        this.user = user
    }
}